CSharp/ArticleProjects/ApplicationInsights/TopNWords/Job.cs

```diff
@@ -0,0 +1,337 @@
+                    unboundJob.Id = topNWordsConfiguration.JobId;
+                    unboundJob.PoolInformation = new PoolInformation() { PoolId = topNWordsConfiguration.PoolId };
+
+                    // Commit Job to create it in the service
+                    unboundJob.Commit();
+
+                    // create file staging objects that represent the executable and its dependent assembly to run as the task.
+                    // These files are copied to every node before the corresponding task is scheduled to run on that node.
+                    FileToStage topNWordExe = new FileToStage(TopNWordsExeName, stagingStorageAccount);
+                    FileToStage storageDll = new FileToStage(StorageClientDllName, stagingStorageAccount);
+
+                    // Upload application insights assemblies
+                    FileToStage applicationInsightsConfig = new FileToStage(AIConfig, stagingStorageAccount);
+                    FileToStage applicationInsightsDll = new FileToStage(AIDllName, stagingStorageAccount);
+                    FileToStage applicationInsightsCollectorDll = new FileToStage(AIDependencyCollectorName, stagingStorageAccount);
+                    FileToStage applicationInsightsInterceptorDll = new FileToStage(AIInterceptDllAgentName, stagingStorageAccount);
+                    FileToStage applicationInsightsPerfCounterCollectorDll = new FileToStage(AIPerfCounterCollectorName, stagingStorageAccount);
+                    FileToStage applicationInsightsServerTelemetryDll = new FileToStage(AIServerTelemetryName, stagingStorageAccount);
+                    FileToStage applicationInsightsWindowsServerDll = new FileToStage(AIWindowsServerName, stagingStorageAccount);
+                    FileToStage batchApplicationInsightsTelemetryInitializerDll = new FileToStage(BatchAITelemetryInitializerName, stagingStorageAccount);
+
+                    // In this sample, the input data is copied separately to Storage and its URI is passed to the task as an argument.
+                    // This approach is appropriate when the amount of input data is large such that copying it to every node via FileStaging
+                    // is not desired and the number of tasks is small since a large number of readers of the blob might get throttled
+                    // by Storage which will lengthen the overall processing time.
+                    //
+                    // You'll need to observe the behavior and use published techniques for finding the right balance of performance versus
+                    // complexity.
+
+                    string[] documents = Directory.GetFiles(topNWordsConfiguration.DocumentsRootPath);
+                    List<string> documentUris = new List<string>();
+                    foreach (string document in documents)
+                    {
+                        documentUris.Add(UploadFileToCloudBlob(accountSettings, document));
```

Comment:
Weird that you are using `FileToStage` in some cases but not others. What's the reasoning?